portfolio/src/main/webapp/comments.js

```diff
@@ -13,24 +13,120 @@
 // limitations under the License.
 
 /**
- * Fetches comments from the server and adds it to the DOM.
+ * An object to manage the query string of a URL
+ * @const {URLSearchParams}
  */
-function getComments() {
-  fetch('/comments').then(response => response.json()).then((comments) => {
-    const commentsContainer = document.getElementById('comments-container');
+const GET_PARAMS = new URLSearchParams ({
+  imageId: '',
+  quantity: '5',
+  dateOrder: 'true'
+});
 
-    commentsContainer.innerHTML = '';
-    for (const comment of comments) {
-      commentsContainer.appendChild(createListElement(comment));
+/**
+ * Delete all the comments
+ */
+function deleteAllComments(imageId) {
+  const postParams = new URLSearchParams();
+  postParams.append('imageId', imageId);
+
+  fetch('/delete-comments', {method: 'POST', body: postParams}).then(response => {
+    if (response.status === 200) {
+      getComments();
+    } else {
+      alert('Sorry, we could not process your request. ' +
+          'Reload the page and try again.');
     }
   });
 }
 
-/** Creates an <li> element containing text. */
-function createListElement(text) {
-  const liElement = document.createElement('li');
-  liElement.innerText = text;
-  return liElement;
+/**
+ * Fetches comments from the server and adds them to the DOM.
+ */
+function getComments() {
+  fetch(`/list-comments?${GET_PARAMS.toString()}`).then((response) => {
+    if (response.status === 200) {
+      response.json().then((comments) => {
+        const commentsContainer = document.querySelector('#comments-container');
+        
+        removeNodeChildren(commentsContainer);
+        addCommentsToDom(comments, commentsContainer);
+      });
+    }
+    else {
+      alert('Sorry, we could not process your request. ' +
+          'Reload the page and try again.');
+    }
+  });
+}
+
+/**
+ * Remove children of a HTML node
+ * @param {Object} node The HTML node, element.
+ */
+function removeNodeChildren(node) {
+  while (node.firstChild) {
+    node.removeChild(node.lastChild);
+  }
 }
 
-document.body.onload = getComments;
\ No newline at end of file
+/**
+ * Adds comments to a container using a template
+ * @param {!Array<{Object}>} comments The comments to add
+ * @param {Object} container The container to place comments inside
+ */
+function addCommentsToDom(comments, container) {
+  const commentTemplate = document.querySelector('#comment-card');
+
+  for (const comment of comments) {
+    const newComment = commentTemplate.content.cloneNode(true);
+
+    newComment.querySelector('h4').textContent = comment.username;
+    newComment.querySelector('span').textContent = comment.date;
+    newComment.querySelector('p').textContent = comment.comment;
+
+    container.appendChild(newComment);
+  }
+}
+
+/**
+ * Set up functions for when comment section events are triggered
+ */
+function setUpComments() {
```

Comment:
```suggestion
function init() {
```
These types of functions are generally called init() or initialize()

Reply:
I see, so just init() or initComments()?

Reply:
Yup, either sounds good to me :)